blockchain/chain_sync/src/sync.rs

```diff
@@ -45,29 +46,13 @@ use std::convert::{TryFrom, TryInto};
 use std::sync::Arc;
 use vm::TokenAmount;
 
-#[derive(PartialEq, Debug, Clone)]
-/// Current state of the ChainSyncer
-pub enum SyncState {
-    /// Initial state, validating data structures and local chain
-    Init,
-
-    /// Bootstrap to the network, and acquire a secure enough set of peers
-    Bootstrap,
-
-    /// Syncing to checkpoint (using BlockSync for now)
-    Checkpoint,
-
-    /// Receive new blocks from the network and sync toward heaviest tipset
-    Catchup,
-
-    /// Once all blocks are validated to the heaviest chain, follow network
-    /// by receiving blocks over the network and validating them
-    Follow,
-}
-
+/// Struct that handles the ChainSync logic. This handles incoming network events such as
+/// gossipsub messages, Hello protocol requests, as well as sending and receiving BlockSync
+/// messages to be able to do the initial sync.
 pub struct ChainSyncer<DB, TBeacon> {
     /// Syncing state of chain sync
-    state: SyncState,
+    // TODO should be a vector once syncing done async and ideally not wrap each state in mutex.
+    state: Arc<RwLock<SyncState>>,
```

Comment:
I'd rename this to stage since the getter/setter are (get/set)_stage. Or change the method name

Reply:
SyncStage is a different meaning, and there is no methods you are referring to, can you reference what you mean?

Reply:
Oh youre right. I misread, the stage is inside the state. M'bad, carry on :)